services/ui-src/src/components/fields/TextField.tsx

```diff
@@ -70,23 +53,15 @@ export const TextField = ({
   // if should autosave, submit field data to database on blur
   const onBlurHandler = async (event: InputChangeEvent) => {
     const { name, value } = event.target;
-    const willAutosave = shouldAutosave(
-      value,
-      lastAutosaveValue,
-      autosave,
-      userIsStateRep,
-      userIsStateUser
-    );
-    if (willAutosave) {
-      setLastAutosaveValue(displayValue);
-      const submissionValue = await validateAndSetValue(form, name, value, "");
-      form.setValue(name, submissionValue, { shouldValidate: true });
-      const reportKeys = createReportKeys(report?.id, state);
-      const dataToWrite = createDataToWrite(
-        { [name]: submissionValue },
-        full_name
-      );
-      await updateReport(reportKeys, dataToWrite);
+    if (autosave) {
+      const fields = [{ name, value, hydrationValue, defaultValue }];
+      const reportArgs = { id: report?.id, updateReport };
+      const user = {
+        userName: full_name,
+        state,
+        isAuthorizedUser: !!(userIsStateRep || userIsStateUser),
+      };
+      await autosaveFieldData({ form, fields, report: reportArgs, user });
```

Comment:
Note: All the autosave stuff in each field file slims down to just this: creating the args that go into the `autosaveFieldData` method.